CorsixTH/Lua/hospitals/player_hospital.lua

```diff
@@ -442,13 +443,39 @@ function PlayerHospital:announceRepair(room)
   if sound then self.world.ui:playAnnouncement(sound, AnnouncementPriority.Critical) end
 end
 
+function PlayerHospital:onSpawnVIP()
+  self.announce_vip = self.announce_vip + 1
+end
+
 --! Called at the end of each day.
 function PlayerHospital:onEndDay()
   -- Advise the player.
   if self:hasStaffedDesk() then
     self:dailyAdviceChecks()
   end
 
+  -- check if we still have to announce VIP visit
+  if self.announce_vip > 0 then
+    -- check if the VIP is in the building yet
+    for _, e in ipairs(self.world.entities) do
+      if e.humanoid_class == "VIP" and e.announced == false and
+          self:isInHospital(e.tile_x, e.tile_y) then
+        -- play VIP arrival sound and show tooltips
+        local ui = self.world.ui
+        -- there is also vip008 which announces a man from the ministry
+        ui:playRandomAnnouncement({ "vip001.wav", "vip002.wav", "vip003.wav",
+            "vip004.wav", "vip005.wav" }, AnnouncementPriority.High)
```

Comment:
Some changes I made to the announcer drops duplicate announcements out the queue. But the realistic likelihood of that happening is very low. Yet, I see no issue to hold this pr up.